src/elements/helpers/Preview.tsx

```diff
@@ -40,23 +85,45 @@ export const Preview = ({
     }
   };
 
+  const listenToMessages = () => {
+    window.addEventListener("message", (message) => {
```

Comment:
We need to be a bit careful here, we are adding event listeners that never being cleaned up.